iree/compiler/Dialect/Flow/Transforms/FusionOfTensorOps.cpp

```diff
@@ -79,12 +79,14 @@ struct FusionOfTensorOpsPass
                           consumer.getOwner()->operand_end());
           if (operands.size() >= kIreeMaxOperandCount) return false;
 
-          llvm::SmallDenseSet<Operation *, 4> numUsers;
-          for (Operation *user : producer.getUsers()) {
-            if (isa<linalg::GenericOp>(user)) continue;
-            numUsers.insert(user);
-          }
-          return numUsers.empty();
+          // Only fuse if it has a single linalg generic user. It is a
+          // simplistic heuristic to avoid duplicating ops that may be
+          // expensive.
+          // TODO: Add a cost model to allow ops to be duplicated.
+          if (std::distance(producer.getUsers().begin(),
```

Comment:
I think you can use `llvm::hasSingleElement(producer.getUsers())`

Reply:
done.